lib/client/client.go

```diff
@@ -288,3 +316,95 @@ func (c *Client) list(metadata unversioned.ResourceMetadata, helper conversionHe
 
 	return nil
 }
+
+// ensureGlobalFelixConfig ensures the specific global felix configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalFelixConfig(key string, defaultValue interface{}) error {
+	kv := &model.KVPair{
+		Key:   model.GlobalConfigKey{Name: key},
+		Value: fmt.Sprintf("%v", defaultValue),
+	}
+	if _, err := c.backend.Create(kv); err == nil {
+		log.WithField(key, defaultValue).Info("Assigned default global Felix config")
+	} else {
+		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
+			log.WithField(key, defaultValue).WithError(err).Warnf("Failed to set global Felix config")
+			return err
+		}
+		log.WithField(key, defaultValue).Info("Using previously configured global Felix config")
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures the specific global BGP configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalBGPConfig(key string, defaultValue interface{}) error {
```

Comment:
Maybe - it's a little more bulky but perhaps if I'm removing a bunch of the defaults then it's not a lot of code.